src/olympia/blocklist/models.py

```diff
@@ -256,49 +257,65 @@ def can_user_signoff(self, signoff_user):
-                for block in self.to_block)
+        return all(
+            (lambda du: du <= threshold and du)(block['average_daily_users'])
+            for block in self.to_block)
+
+    def has_version_changes(self):
+        block_ids = [block['id'] for block in self.to_block]
+
+        has_new_blocks = any(not id_ for id_ in block_ids)
+        blocks_with_version_changes_qs = Block.objects.filter(
+            id__in=block_ids).exclude(
+                min_version=self.min_version, max_version=self.max_version)
 
-        if self.signoff_state == self.SIGNOFF_PENDING and not any_unsafe():
-            self.update(signoff_state=self.SIGNOFF_NOTNEEDED)
+        return has_new_blocks or blocks_with_version_changes_qs.exists()
+
+    def update_if_signoff_not_needed(self):
+        is_pending = self.signoff_state == self.SIGNOFF_PENDING
+        add_action = self.action == self.ACTION_ADDCHANGE
+        if (
+            (is_pending and self.all_adu_safe()) or
+            (is_pending and add_action and not self.has_version_changes())
+        ):
+            self.update(signoff_state=self.SIGNOFF_AUTOAPPROVED)
 
     @property
     def is_submission_ready(self):
         """Has this submission been signed off, or sign-off isn't required."""
         return (
-            self.signoff_state == self.SIGNOFF_NOTNEEDED or (
+            self.signoff_state == self.SIGNOFF_AUTOAPPROVED or (
                 self.signoff_state == self.SIGNOFF_APPROVED and
                 self.can_user_signoff(self.signoff_by)
             )
         )
 
-    def _serialize_blocks(self, blocks):
+    def _serialize_blocks(self):
+
         def serialize_block(block):
             return {
                 'id': block.id,
                 'guid': block.guid,
-                'average_daily_users': (
-                    block.addon.average_daily_users if block.addon else -1),
+                'average_daily_users':
+                    block.addon.average_daily_users if block.addon else -1,
             }
 
-        return [serialize_block(block) for block in blocks]
+        processed = self.process_input_guids(
+            self.input_guids,
+            self.min_version,
+            self.max_version,
+            load_full_objects=False,
+            filter_existing=(self.action == self.ACTION_ADDCHANGE))
+        return [
+            serialize_block(block) for block in processed.get('blocks', [])]
 
     def save(self, *args, **kwargs):
         if self.input_guids and not self.to_block:
-            processed = self.process_input_guids(
-                self.input_guids,
-                self.min_version,
-                self.max_version,
-                load_full_objects=False,
-                filter_existing=(self.action == self.ACTION_ADDCHANGE))
             # serialize blocks so we can save them as JSON
-            self.to_block = self._serialize_blocks(processed.get('blocks', []))
+            self.to_block = self._serialize_blocks()
```

Comment:
This change to move more code to `self._serialize_blocks` was a refactor for an aborted implementation but I preferred the cleaner `save` method so kept it.